File: BackEnd/src/Model/Entities/Unitat.java
package hospiolot.model.business.entities;

import hospiolot.model.business.entities.Entity;

public class Unitat extends Entity {

    private String nom;

   
    private long idCap;

    public Unitat(String nom) {
        this.nom = nom;
    }
    
    public Unitat(String nom, long idCap) {
        this.nom = nom;
        this.idCap = idCap;
       
    }
    
     public Unitat(Unitat u) {
        this(u.getNom(),u.getIdCap());
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }
    
     public long getIdCap() {
        return idCap;
    }

    public void setIdCap(long idCap) {
        this.idCap = idCap;
    }

    @Override
    public String toString() {
        return nom ;
    }

}

File: BackEnd/src/Model/Entities/Unitat.ts
class Unitat {

    private nom:string;

   
    private idCap:number;

    constructor(nom?: string, idCap?: number, unitat?: Unitat) {
        if (unitat) {
            this.nom = unitat.nom;
            this.idCap = unitat.idCap;
        }else if (nom && idCap) { 
            this.nom = nom;
            this.idCap = idCap;
        }else if(nom){
            this.nom = nom;
        }
    }

    public getNom():string {
        return this.nom;
    }

    public setNom(nom:string):void {
        this.nom = nom;
    }
    
     public getIdCap():number {
        return this.idCap;
    }

    public setIdCap(idCap:number):void {
        this.idCap = idCap;
    }

    public toString():string {
        return this.nom ;
    }

}

export = Unitat;